### PR TITLE
Add information about link entities

### PR DESCRIPTION
**Pull request by thomasyu888**

### CODE REVIEW

File: articles/links.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+<style>
+#image {
+    width: 50%;
+}
+</style>
+
+
+# Links
+Synapse `Links` is a feature designed to provide users the ability to create a link to any file, table, folder, or project on Synapse.
```
Comment (kdaily) [inline articles/links.md]: tense is wrong.

Synapse `Links` provide users the ability...

File: articles/links.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+
+# Links
+Synapse `Links` is a feature designed to provide users the ability to create a link to any file, table, folder, or project on Synapse. 
+
+
+
+## Creating a Link
+
+There are currently only ways to create links through the web and Python client.
```
Comment (kdaily) [inline articles/links.md]: Move this into an `R` tab.

File: articles/links.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+#targetId is the synapse id of the file, table, etc that you want to link
+#targetVersion is optional, if no version is defined, the link will always point to the newest version
+#parent is the folder or project where you want to link to exist
+syn.store(linkEnt)
+
+{% endhighlight %}
+{% endtab %}
+
+{% tab Web %}
+Navigate to the file, table, folder or project you want to save a link of. Click Tools and Save Link to .
```
Comment (kdaily) [inline articles/links.md]: extra space at end

File: articles/links.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+<br>
+<img id="image" src="/assets/images/save_link_to_file.png">
+<br>
+
+Select a folder or project that you want to save the link to (This feature is more like bookmarking the file to a specific location)
```
Comment (kdaily) [inline articles/links.md]: no ending punctuation

File: articles/downloading_data.md [hunk 5]
```diff
@@ -102,6 +102,29 @@ entity <- synGet("syn3260973", version=1)
 
 See [versioning](http://docs.synapse.org/articles/versioning.html) for more details.
 
+### Links
+
+There are multiple ways of downloading a link in Synapse.  Without specifying the followLink parameter, you will only retrieve the link itself without downloading wherever the link leads to.  There is currently not a way to follow the link in R.
```
Comment (kdaily) [inline articles/downloading_data.md]: what are the multiple ways?

Why is there not a way to follow the link in R? We have `synapser` now.

File: articles/links.md (hunk 2 above)
Comment (kdaily) [reply articles/links.md]: But we should have a way in `synapser`, so I think this isn't current.

**Comment by thomasyu888** [comment]
Are we updating all of the docs to include `synapser` now?

**Comment by kdaily** [comment]
@thomasyu888 all docs on docs.synapse.org use `synapser` now, correct!

File: articles/downloading_data.md [hunk 6]
```diff
@@ -102,6 +102,34 @@ entity <- synGet("syn3260973", version=1)
 
 See [versioning](http://docs.synapse.org/articles/versioning.html) for more details.
 
+### Links
+
+When you click on a link online, it will automatically direct you to the linked entity.  The `followLink` parameter will have to be specified when using the analytical clients or you will only retrieve the link itself without downloading the linked entity.
```
Comment (kdaily) [inline articles/downloading_data.md]: online? On the Synapse web site?

Comment (kdaily) [reply articles/downloading_data.md]: 'automatically direct' -> 'redirect'

Comment (kdaily) [reply articles/downloading_data.md]: 'analytical' -> 'programmatic'

File: articles/downloading_data.md [hunk 7]
```diff
@@ -102,6 +102,34 @@ entity <- synGet("syn3260973", version=1)
...
+linkEnt = syn.get("syn1234")
+entity = syn.get("syn1234", followLink=True)
+{% endhighlight %}
+{% endtab %}
+{% tab R %}
+{% highlight r %}
+library(synapser)
+synLogin()
+linkEnt = synGet("syn1234")
+entity = synGet("syn1234",followLink=T)
```
Comment (kdaily) [inline articles/downloading_data.md]: `followLink=TRUE`

Comment (kdaily) [reply articles/downloading_data.md]: fix spacing around comma

File: articles/links.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    width: 50%;
+}
+#link {
+    width: 100%;
+}
+</style>
+
+
+# Links
+Synapse `Links` provide users the ability to create a link to any file, table, folder, or project on Synapse.
```
Comment (kdaily) [inline articles/links.md]: File, Table, Folder, or Project

File: articles/links.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
+---
+title: Links
+layout: article
+excerpt: The basics of creating, and modifying links on Synapse.
```
Comment (kdaily) [inline articles/links.md]: why just the basics? we document everything.

File: articles/links.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+{% tabs %}
+
+{% tab Python %}
+{% highlight python %}
+import synapseclient
+syn = synapseclient.login()
+
+# Add a local file to an existing project (syn12345) on Synapse
+linkEnt = synapseclient.Link(targetId="syn12345",targetVersion=1, parent="syn2345")
```
Comment (kdaily) [inline articles/links.md]: spacing around commas

File: articles/links.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+#targetId is the synapse id of the file, table, etc that you want to link
+#targetVersion is optional, if no version is defined, the link will always point to the newest version
+#parent is the folder or project where you want to link to exist
+syn.store(linkEnt)
```
Comment (kdaily) [inline articles/links.md]: `linkEnt = syn.store...`

File: articles/links.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+{% endhighlight %}
+{% endtab %}
+
+{% tab R %}
+{% highlight r %}
+library(synapser)
+synLogin()
+
+# Add a local file to an existing project (syn12345) on Synapse
+linkEnt <- Link(targetId="syn12345",targetVersion=1, parent="syn2345")
```
Comment (kdaily) [inline articles/links.md]: fix spacing around comma

File: articles/links.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+#targetId is the synapse id of the file, table, etc that you want to link
+#targetVersion is optional, if no version is defined, the link will always point to the newest version
+#parent is the folder or project where you want to link to exist
```
Comment (kdaily) [inline articles/links.md]: all these comments should go before the line they refer to

File: articles/links.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+{% tab Python %}
+{% highlight python %}
+import synapseclient
+syn = synapseclient.login()
+
+# Add a local file to an existing project (syn12345) on Synapse
+linkEnt = synapseclient.Link(targetId="syn12345",targetVersion=1, parent="syn2345")
+#targetId is the synapse id of the file, table, etc that you want to link
+#targetVersion is optional, if no version is defined, the link will always point to the newest version
+#parent is the folder or project where you want to link to exist
```
Comment (kdaily) [inline articles/links.md]: all comments here should go before line they refer to

Comment (kdaily) [reply articles/links.md]: comments should start with `# there's a space here`

File: articles/links.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+<br>
+<img id="toobig" src="/assets/images/save_link_to_file.png">
+<br>
+
+Select a folder or project that you want to save the link to (This feature is more like bookmarking the file to a specific location).
+<br>
+<img id="image" src="/assets/images/link_to_parent.png">
+<br>
+
+The final result looks like.
```
Comment (kdaily) [inline articles/links.md]: end in colon

**Pull request merged**